### PR TITLE
[CI] Fix the download step

### PR DESCRIPTION
**Pull request by mandel-macaque**
Allow to download the artifacts from the build that triggered the pipeline.

### CODE REVIEW

**Review by stephen-hawley — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

File: tools/devops/automation/templates/governance/apiscan.yml [hunk 1]
```diff
@@ -27,12 +27,27 @@ steps:
...
+    displayName: Download artifacts
+    artifact: not-signed-package
+
+  - pwsh: |
+      $source = "$(Pipeline.Workspace)/macios/not-signed-package"
+      $destination = "$(Build.SourcesDirectory)/not-signed-package"
+      New-Item -ItemType Directory -Force -Path $destination 
+      Write-Host "Moving content from $source to $destination"
+      # move all the files from the source to the destination
+      Get-ChildItem -Path $source -Recurse -File | Move-Item -Destination $destination
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/governance/apiscan.yml:41]: I don't think this works if there are files in subdirectories.

Can't you just move/copy `$source` to `$destination`?

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: tools/devops/automation/templates/governance/apiscan.yml (hunk 1 above)
Comment (mandel-macaque) [reply tools/devops/automation/templates/governance/apiscan.yml:41]: The script is not copying the files as they are, it is flattening the directory and copying all the files in it. So it is nota move or a copy. Is a flattening and a copy.

Comment (rolfbjarne) [reply tools/devops/automation/templates/governance/apiscan.yml:41]: Why would that be needed here and not in the `${{ else }}` section below?

Can you add a comment explaining why flattening is needed?

Comment (mandel-macaque) [reply tools/devops/automation/templates/governance/apiscan.yml:41]: yes, the issue is that the download task downloads under ${pipe-reference}/artifact while the other task does download to the desired path. So we need to do that. I'll add a comment.

Comment (mandel-macaque) [reply tools/devops/automation/templates/governance/apiscan.yml:45]: You can find a successful run here: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=10369133&view=results

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**